### PR TITLE
Fix file paste: cache file info

### PR DESCRIPTION
**Pull request by ellatrix**
## Description
See #6960. This change is causing errors for me when copy pasting images in Chrome.

## How has this been tested?
* Copy paste an image into Gutenberg.
* Drap and drop an image into Gutenberg.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation.

### CODE REVIEW

File: packages/blob/src/index.js [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
+	};
 
 	return url;
 }
 
 export function getBlobByURL( url ) {
 	if ( cache[ url ] ) {
-		return Promise.resolve( cache[ url ] );
+		const { blob, name, lastModified, type } = cache[ url ];
+		const file = new File( [ blob ], name, { lastModified, type } );
```
Comment (jorgefilipecosta) [inline packages/blob/src/index.js]: Would caching new File( [ blob ], name, { lastModified, type } ); directly in cache[ url ] be an option?

**Review by jorgefilipecosta — COMMENTED** [review]
This seems to solve the problem happening on copy paste. When using drag&drop the images continue to keep their file names. The code becomes better with this changes an awesome advantage of now using createBlobURL instead of createObjectURL.
Thank you a lot for this fix!

File: packages/blob/src/index.js (hunk 1 above)
Comment (jorgefilipecosta) [reply packages/blob/src/index.js]: e.g. in createBlobURL do cache[ url ] = new File( [ blob ], name, { lastModified, type } );

Comment (iseulde) [reply packages/blob/src/index.js]: Sure, sounds good

**Comment by iseulde** [comment]
I can't recall why the `fetch` was added in #2792... Perhaps a fall back if there's no cache left or it's somehow copy pasted, but I don't see a specific use case after searching a while. The problem with this line is that we can't set a mime type and it won't go through anyway, so I'm going to remove it.

**Comment by iseulde** [comment]
@jorgefilipecosta Does everything still work for you with the latest change?

File: packages/blob/src/index.js [hunk 2]
```diff
@@ -1,27 +1,41 @@
...
+ *
+ * @return {File} The file for the blob URL.
+ */
 export function getBlobByURL( url ) {
-	if ( cache[ url ] ) {
-		return Promise.resolve( cache[ url ] );
-	}
-
-	return fetch( url ).then( ( response ) => response.blob() );
+	return cache[ url ];
```
Comment (jorgefilipecosta) [inline packages/blob/src/index.js]: Before we verified if the file exists in the cache, is it safe to assume that for all blob url's used here we used createBlobURL before? In that case, I think we should specify in the comments that this function only works for blubUrls created by createBlobURL.

Comment (iseulde) [reply packages/blob/src/index.js]: Sure, that hasn't changed. I'll add a comment.

**Review by jorgefilipecosta — APPROVED** [review]
I tested drag&drop with single & multiple images the names were kept with success.
Copy & paste for image files works great the names are not kept but I think it is not possible to access the original file name during a copy paste.

I think we should add comment in getBlobByURL to specify it only works for blob urls created by createBlobUR. But besides that everything looks good to me.
Nice work, thank you for this PR 👍

**Pull request merged**